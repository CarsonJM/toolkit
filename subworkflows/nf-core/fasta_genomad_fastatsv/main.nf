// Import modules
include { GENOMAD_DOWNLOAD  } from '../../../modules/nf-core/genomad/download/main'
include { GENOMAD_ENDTOEND  } from '../../../modules/nf-core/genomad/endtoend/main'

workflow FASTA_GENOMAD_FASTATSV {
    take:
    fasta_gz    // [ [ meta ], fasta.gz ]   , assemblies/genomes (mandatory)
    genomad_db  // [ genomad_db ]           , genomad database directory (optional)

    main:
    ch_versions = Channel.empty()

    // if genomad_db exists, skip GENOMAD_DOWNLOAD
    if (genomad_db){
        ch_genomad_db = genomad_db
    } else {
        //
        // MODULE: download geNomad database
        //
        GENOMAD_DOWNLOAD(
        )
        ch_genomad_db   = GENOMAD_DOWNLOAD.out.genomad_db
        ch_versions     = ch_versions.mix(GENOMAD_DOWNLOAD.out.versions)
    }

    //
    // MODULE: Run geNomad end-to-end
    //
    GENOMAD_ENDTOEND(
        fasta_gz,
        ch_genomad_db
    )
    ch_versions = ch_versions.mix(GENOMAD_ENDTOEND.out.versions)

    emit:
    plasmid_fasta_gz    = GENOMAD_ENDTOEND.out.plasmid_fasta    // [ [ meta ], plasmid.fna.gz ]
    plasmid_faa_gz      = GENOMAD_ENDTOEND.out.plasmid_proteins // [ [ meta ], plasmid.faa.gz ]
    plasmid_tsv         = GENOMAD_ENDTOEND.out.plasmid_summary  // [ [ meta ], plasmid_summary.tsv ]
    virus_fasta_gz      = GENOMAD_ENDTOEND.out.virus_fasta      // [ [ meta ], virus.fna.gz ]
    virus_faa_gz        = GENOMAD_ENDTOEND.out.virus_proteins   // [ [ meta ], virus.faa.gz ]
    virus_tsv           = GENOMAD_ENDTOEND.out.virus_summary    // [ [ meta ], virus_summary.tsv ]
    protein_faa_gz      = GENOMAD_ENDTOEND.out.proteins         // [ [ meta ], proteins.faa.gz ]
    genes_tsv           = GENOMAD_ENDTOEND.out.genes            // [ [ meta ], genes.tsv ]
    features_tsv        = GENOMAD_ENDTOEND.out.features         // [ [ meta ], features.tsv ]
    scores_tsv          = GENOMAD_ENDTOEND.out.scores           // [ [ meta ], aggregated_classification.tsv ]
    taxonomy_tsv        = GENOMAD_ENDTOEND.out.taxonomy         // [ [ meta ], taxonomy.tsv ]
    versions            = ch_versions                           // channel: [ versions.yml ]
}
