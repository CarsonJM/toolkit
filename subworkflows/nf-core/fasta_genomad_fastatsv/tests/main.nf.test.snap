{
    "sarscov2 - genome - fasta": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_plasmid.fna.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ],
                    [
                        {
                            "id": "test2"
                        },
                        "test2_plasmid.fna.gz:md5,e0315e8db8ded8fd70b2779f19d66d54"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_plasmid_proteins.faa.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ],
                    [
                        {
                            "id": "test2"
                        },
                        "test2_plasmid_proteins.faa.gz:md5,baa62e6878e0ba772bca4ea4152be28b"
                    ]
                ],
                "10": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_taxonomy.tsv:md5,44c6cf5ff4c836f66648457ae8deb0d5"
                    ],
                    [
                        {
                            "id": "test2"
                        },
                        "test2_taxonomy.tsv:md5,22081fd09bee75003b5aac12f05b7a29"
                    ]
                ],
                "11": [
                    "versions.yml:md5,05c77f0513d8d9f8e7634b0f4f30ae61",
                    "versions.yml:md5,6c264fbc929fe9f57b97eac5e082c2bc",
                    "versions.yml:md5,6c264fbc929fe9f57b97eac5e082c2bc"
                ],
                "2": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_plasmid_summary.tsv:md5,8c4ddaa8a90da779c11537be0fddb799"
                    ],
                    [
                        {
                            "id": "test2"
                        },
                        "test2_plasmid_summary.tsv:md5,951e9110c265af707b2c65500a7acacd"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_virus.fna.gz:md5,483f4a5dfe60171c86ee9b7e6dff908b"
                    ],
                    [
                        {
                            "id": "test2"
                        },
                        "test2_virus.fna.gz:md5,2a915977fdd3099acc5c851c919271f0"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_virus_proteins.faa.gz:md5,6850bd46dcb921e5cfb23bfbb8fcc12b"
                    ],
                    [
                        {
                            "id": "test2"
                        },
                        "test2_virus_proteins.faa.gz:md5,468b48e3ee919305b1a5bad3654b5ac1"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_virus_summary.tsv:md5,923b36db276dfee18ac1f1b57cae617e"
                    ],
                    [
                        {
                            "id": "test2"
                        },
                        "test2_virus_summary.tsv:md5,e2f291c05ebe6cbeaca7a30cb845eb04"
                    ]
                ],
                "6": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_proteins.faa.gz:md5,6850bd46dcb921e5cfb23bfbb8fcc12b"
                    ],
                    [
                        {
                            "id": "test2"
                        },
                        "test2_proteins.faa.gz:md5,63d0544a5d8e1448d375e09329eec1b7"
                    ]
                ],
                "7": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_genes.tsv:md5,f84d8a9fce3c92c49e78672bea16561f"
                    ],
                    [
                        {
                            "id": "test2"
                        },
                        "test2_genes.tsv:md5,2cd0f7b98a3c880e7eea765aa821ff19"
                    ]
                ],
                "8": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_features.tsv:md5,868804cf1857d3c7e129086ec747c77c"
                    ],
                    [
                        {
                            "id": "test2"
                        },
                        "test2_features.tsv:md5,2de71bfdaa71a6a6ced6dc5160c6378c"
                    ]
                ],
                "9": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_aggregated_classification.tsv:md5,702aead49cca96c98e43c5f032f60b26"
                    ],
                    [
                        {
                            "id": "test2"
                        },
                        "test2_aggregated_classification.tsv:md5,52472567bef1698395d7afef49bd0350"
                    ]
                ],
                "features_tsv": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_features.tsv:md5,868804cf1857d3c7e129086ec747c77c"
                    ],
                    [
                        {
                            "id": "test2"
                        },
                        "test2_features.tsv:md5,2de71bfdaa71a6a6ced6dc5160c6378c"
                    ]
                ],
                "genes_tsv": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_genes.tsv:md5,f84d8a9fce3c92c49e78672bea16561f"
                    ],
                    [
                        {
                            "id": "test2"
                        },
                        "test2_genes.tsv:md5,2cd0f7b98a3c880e7eea765aa821ff19"
                    ]
                ],
                "plasmid_faa_gz": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_plasmid_proteins.faa.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ],
                    [
                        {
                            "id": "test2"
                        },
                        "test2_plasmid_proteins.faa.gz:md5,baa62e6878e0ba772bca4ea4152be28b"
                    ]
                ],
                "plasmid_fasta_gz": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_plasmid.fna.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ],
                    [
                        {
                            "id": "test2"
                        },
                        "test2_plasmid.fna.gz:md5,e0315e8db8ded8fd70b2779f19d66d54"
                    ]
                ],
                "plasmid_tsv": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_plasmid_summary.tsv:md5,8c4ddaa8a90da779c11537be0fddb799"
                    ],
                    [
                        {
                            "id": "test2"
                        },
                        "test2_plasmid_summary.tsv:md5,951e9110c265af707b2c65500a7acacd"
                    ]
                ],
                "protein_faa_gz": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_proteins.faa.gz:md5,6850bd46dcb921e5cfb23bfbb8fcc12b"
                    ],
                    [
                        {
                            "id": "test2"
                        },
                        "test2_proteins.faa.gz:md5,63d0544a5d8e1448d375e09329eec1b7"
                    ]
                ],
                "scores_tsv": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_aggregated_classification.tsv:md5,702aead49cca96c98e43c5f032f60b26"
                    ],
                    [
                        {
                            "id": "test2"
                        },
                        "test2_aggregated_classification.tsv:md5,52472567bef1698395d7afef49bd0350"
                    ]
                ],
                "taxonomy_tsv": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_taxonomy.tsv:md5,44c6cf5ff4c836f66648457ae8deb0d5"
                    ],
                    [
                        {
                            "id": "test2"
                        },
                        "test2_taxonomy.tsv:md5,22081fd09bee75003b5aac12f05b7a29"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,05c77f0513d8d9f8e7634b0f4f30ae61",
                    "versions.yml:md5,6c264fbc929fe9f57b97eac5e082c2bc",
                    "versions.yml:md5,6c264fbc929fe9f57b97eac5e082c2bc"
                ],
                "virus_faa_gz": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_virus_proteins.faa.gz:md5,6850bd46dcb921e5cfb23bfbb8fcc12b"
                    ],
                    [
                        {
                            "id": "test2"
                        },
                        "test2_virus_proteins.faa.gz:md5,468b48e3ee919305b1a5bad3654b5ac1"
                    ]
                ],
                "virus_fasta_gz": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_virus.fna.gz:md5,483f4a5dfe60171c86ee9b7e6dff908b"
                    ],
                    [
                        {
                            "id": "test2"
                        },
                        "test2_virus.fna.gz:md5,2a915977fdd3099acc5c851c919271f0"
                    ]
                ],
                "virus_tsv": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_virus_summary.tsv:md5,923b36db276dfee18ac1f1b57cae617e"
                    ],
                    [
                        {
                            "id": "test2"
                        },
                        "test2_virus_summary.tsv:md5,e2f291c05ebe6cbeaca7a30cb845eb04"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-10T06:47:16.212676086"
    }
}